test/SlowTests/Issues/RavenDB_8044.cs

```diff
@@ -0,0 +1,128 @@
+                {
+                    await CreateDatabaseInCluster(storeA, "A");
+                    await EnsureDatabaseOnNode(storeA, "A");
+
+                    using (var session = storeA.OpenSession())
+                    {
+                        session.Store(new User { Name = "James" });
+                        session.SaveChanges();
+                    }
+
+                    using (var session = storeB.OpenSession())
+                    {
+                        session.Store(new User { Name = "Doe" });
+                        session.SaveChanges();
+                    }
+                }
+            }
+        }
+
+        private async Task EnsureDatabaseOnNode(DocumentStore storeB, string node)
+        {
+            await WaitForValueAsync(async () => await GetMembersCount(storeB, _database), 1);
+
+            var topology = await storeB.Admin.Server.SendAsync(new GetDatabaseTopologyOperation(_database));
+
+            Assert.Equal(node, topology.Members[0]);
+        }
+
+        private static async Task<int> GetMembersCount(IDocumentStore store, string databaseName)
+        {
+            var res = await store.Admin.Server.SendAsync(new GetDatabaseTopologyOperation(databaseName));
+            if (res == null)
+            {
+                return -1;
+            }
+            return res.Members.Count;
+        }
+
+        private RavenServer GetNodeServer(string nodeTag)
+        {
+            return Servers.First(x => x.ServerStore.NodeTag == nodeTag);
+        }
+
+        private DocumentStore GetDocumentStoreForServer(RavenServer server)
+        {
+            var store = new DocumentStore
+            {
+                Urls = server.WebUrls,
+                Database = _database
+            };
+
+            return (DocumentStore)store.Initialize();
+        }
+
+        private async Task CreateDatabaseInCluster(IDocumentStore store, string node)
+        {
+            var record = MultiDatabase.CreateDatabaseDocument(_database);
+
+            record.Topology = new DatabaseTopology
+            {
+                Members = new List<string>
+                {
+                    node
+                }
+            };
+
+            var databaseResult = await store.Admin.Server.SendAsync(new CreateDatabaseOperation(record));
+
+            foreach (var server in Servers)
+            {
+                await server.ServerStore.Cluster.WaitForIndexNotification(databaseResult.RaftCommandIndex);
+            }
+        }
+
+        private async Task DeleteDatabaseInCluster(string databaseName, IDocumentStore store, string fromNode)
+        {
+            var databaseResult = await store.Admin.Server.SendAsync(new DeleteDatabaseOperation(databaseName, true, fromNode));
+
+            foreach (var server in Servers)
+            {
+                await server.ServerStore.Cluster.WaitForIndexNotification(databaseResult.RaftCommandIndex + 1);
```

Comment:
OK, I see. That means `DeleteDatabaseOperation` isn't able to return a raft index which that will guarantee that db is deleted from all nodes, right?